docs/how.rst

```diff
@@ -0,0 +1,256 @@
+=====================
+How it Works
+=====================
+
+This new section will cover general info about how the following things work. 
+This is not a how-to, just a way of connecting the dots so that when people 
+are doing these tasks later, it all makes sense.
+
+XXX Description of the order of events. That is... you pick your webserver and OS on the interactive installation tool. Behind the scenes, certbot knows how to modify the config file and handle the challenge. You can optionally do hooks. You get authenticated and the cert gets installed. A config file gets created so that automatic renewal can happen. After that, you can manage, modify, or delete certs.
+
+
+
+
+.. include:: challenges.rst
+
+
+.. _plugins:
+
+
+Plugins
+=======
+
+The Certbot client supports two types of plugins for
+obtaining and installing certificates: authenticators and installers.
+
+Authenticators are plugins used with the ``certonly`` command to obtain a certificate.
+The authenticator satisfies a challenge to validate that you
+control the domain(s) you are requesting a certificate for, obtains a certificate for the specified
+domain(s), and places the certificate in the ``/etc/letsencrypt`` directory on your
+machine. The authenticator does not install the certificate (it does not edit any of your server's configuration files to serve the
+obtained certificate). If you specify multiple domains to authenticate, they will
+all be listed in a single certificate. To obtain multiple separate certificates
+you will need to run Certbot multiple times.
+
+
+Installers are Plugins used with the ``install`` command to install a certificate.
+These plugins can modify your webserver's configuration to
+serve your website over HTTPS using certificates obtained by certbot.
+
+Plugins that do both can be used with the ``certbot run`` command, which is the default
+when no command is specified. The ``run`` subcommand can also be used to specify
+a combination_ of distinct authenticator and installer plugins.
+
+=========== ==== ==== =============================================================== =============================
+Plugin      Auth Inst Notes                                                           Challenge types (and port)
+=========== ==== ==== =============================================================== ===========================
+apache_     Y    Y    | Automates obtaining and installing a certificate with Apache  tls-sni-01_ (443)
```

Comment:
All of these challenge links now no longer have anything to link to since those sections were broken out into another file.
You should update them to point to the proper new file.
These links are `http-01_`, `dns-01_`, `tls-sni-01_`, `standalone_`, `webroot_`, `apache_`, `nginx_`, and `manual_`